### ISSUE TITLE
Merging the knuu network tests feature branch

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#3557]
We have a branch that was created to help unblock the network tests in knuu, and now we need to merge it. Below is a list of issues that are classified as either needed or nice to have. All issues in the needed section are required to be closed to also close this issue. Please feel free to edit as you see fit @staheri14 @smuu 

### Needed
- [x] #3589 
  - [x]  https://github.com/celestiaorg/celestia-app/pull/3591
- [x] https://github.com/celestiaorg/celestia-app/issues/3437
  - [x] https://github.com/celestiaorg/celestia-app/pull/3541
- [x] https://github.com/celestiaorg/celestia-app/pull/3505
- [x] https://github.com/celestiaorg/celestia-app/pull/3515
- [x] https://github.com/celestiaorg/celestia-app/issues/3588
  - [x]  #3595
- [x] https://github.com/celestiaorg/celestia-app/pull/3612
  - [x] https://github.com/celestiaorg/celestia-app/pull/3415 (its advanced version was merged by #3612)
